src/util/assert.h

```diff
@@ -2,20 +2,29 @@
 
 #include <QtDebug>
 
-static constexpr const char* kDebugAssertPrefix = "DEBUG ASSERT";
-
-#ifdef MIXXX_DEBUG_ASSERTIONS_ENABLED
-inline void mixxx_debug_assert(const char* assertion, const char* file, int line, const char* function) {
-    qCritical("%s: \"%s\" in function %s at %s:%d", kDebugAssertPrefix, assertion, function, file, line);
-}
+#ifdef __OPTIMIZE__
+#ifdef __has_builtin
+#if __has_builtin(__builtin_constant_p)
+#define MIXXX_LINK_ASSERTIONS_ENABLED
+#endif
+#elif __GNUC__ < 10
+#define MIXXX_LINK_ASSERTIONS_ENABLED
 #endif
+#endif
```

Comment:
The code above enables the feature only on optimized builds. 
We can add an extra CMake flag to control it. 
Off for local builds.
Than the contributor is surprised when CI fails. 

Maybe we may merge it and collect experience ...

Reply:
No, the builds should behave consistently and the outcome should always be predictable no matter where and how the code is built.